T7664: support FRR 10.5

hedrok opened the pull request:
## Support FRR 10.5

Support for the following changes:

* Removal of 'connected', 'kernel' and 'table' protocols in 'ip protocol'. Added migration script that just removes commands with these protocols (added config test). Updated smoke tests.
* In segment-routing FRR now doesn't output `block-len`, `node-len` and `func-bits` if they have default value. Updated smoke test accordingly.
* FRR 10.5 changed default value for `bgp reject-as-sets` to on by default. I haven't changed default, just added `no bgp reject-as-sets` in FRR configuration template.


## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [X] Other (please describe): Support FRR 10.5

## Related Task(s)

* https://vyos.dev/T7664

## Related PR(s)

* https://github.com/vyos/vyos-build/pull/1090

## How to test / Smoketest result

With vyos-build PR and this PR all smoke tests (VPP not tested) pass.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [X] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [X] My change requires a change to the documentation: https://docs.vyos.io/en/latest/configuration/system/ip.html need to update protocol list
- [X] I have updated the documentation accordingly: https://github.com/vyos/vyos-documentation/pull/1730

hedrok committed:
bgp: T7664: add 'no bgp reject-as-sets'

In FRR 10.5 'bgp reject-as-sets' became default
(https://frrouting.org/release/10.5.0/):

> BGP rejects AS_SET by default
> Until 10.5.0, it was disabled by default, and since RFC 9774 was
> published, we switched this on by default (to reject).

Add 'no bgp reject-as-sets' when VyOS option reject_as_sets is not set.

hedrok committed:
tests: T7664: segment-routing, remove defaults

In FRR 10.5 block-len, node-len and func-bits are not
shown if they have default values.

Fix test smoketest/scripts/cli/test_protocols_segment-routing.py
accordingly (update expected + change func-bits 16 -> 12 so that they are shown).

Relevant commits in FRR:
44e67d2d3dabdc1492448e6a3c7024bd0309c244
84c807a3b300f4c9fcda55bccd93dcb64716f5d7

hedrok committed:
T7664: ip protocol 'kernel' etc removed

In FRR 10.5 ip protocols

* connected
* kernel
* local
* table
* table-direct

are removed. First three in fb8e399e4f66d09780f176fbecb99168089e64eb,
table* in 7fd030504be060387694e8a2af7f19ddd7dee39d.

In `ip protocols`, `ipv6 protocols` and `vrf`  VyOS supports

* connected
* kernel
* table

Remove these from CLI, add migration scripts, update tests.

c-po reviewed (changes requested):
Can you please create or extend `configtest` from VyOS 1.4 which makes use of the now no longer supported `redistribute` statements for both the default and any other VRF? This way we have a build in test that the CLI entries in question will be migrated away.

See for details https://github.com/vyos/vyos-1x/blob/current/smoketest/README.md

hedrok commented:
@c-po Thanks, I've added config test, and it showed one mistake (I've fixed that too).

c-po commented on smoketest/configs/basic-system-ip-protocol:
```diff
@@ -0,0 +1,436 @@
+}
+/* Warning: Do not remove the following line. */
+/* === vyatta-config-version: "broadcast-relay@1:cluster@1:config-management@1:conntrack-sync@1:conntrack@1:dhcp-relay@2:dhcp-server@5:dns-forwarding@1:firewall@5:ipsec@5:l2tp@1:mdns@1:nat@4:ntp@1:pptp@1:qos@1:quagga@6:snmp@1:ssh@1:system@27:vrf@3:vrrp@2:wanloadbalance@3:webgui@1:webproxy@1:webproxy@2:zone-policy@1" === */
+/* Release version: 1.2.6 */
```
This version line is wrong as VRF support for any protocol was only intruduced recently in VyOS 1.4

hedrok replied:
Thank you, I've updated it.
I've edited previous file (with '1.2.6' version) manually to get result I needed - I've just added sections with protocols and added versions of system and vrf.
The updated file is file saved by VyOS 1.4.3 without modifications.
To create it I've:
1. Loaded `basic-vyos` configuration on VyOS 1.4.3
2. Ran commands:
```
set vrf name red table 1234
set policy route-map stub
set system ip protocol any       route-map stub
set system ip protocol babel     route-map stub
set system ip protocol bgp       route-map stub
set system ip protocol connected route-map stub
...
set system ip protocol table     route-map stub
set system ipv6 ...
set vrf name red ip protocol ...
set vrf name red ipv6 protocol ...
```
3. Saved configuration file.

c-po commented on smoketest/configs/basic-system-ip-protocol:
```diff
@@ -0,0 +1,388 @@
+            port "8888"
+        }
+    }
+}
```
I see no benefit in a load-balancing testcase here - we should declutter it
```suggestion
```

c-po commented on smoketest/configs/basic-system-ip-protocol:
```diff
@@ -0,0 +1,388 @@
+        listen-address "192.168.0.1"
+        port "22"
+    }
+}
```
I see no benefit in testing with those services enabled. Sometimes it is also a good idea to test with less config elements and remove all services.

```suggestion
```

hedrok commented:
Thanks for decluttering, as I've described, I just took `basic-vyos`, added needed configuration and changed nothing else as I'm not sure which parts of `basic-vyos` are needed.

c-po reviewed (approved):
Manually tested multiple times. All smoketests now pass as expected.
Requires FRR patches which are currently pending upstream.

sever-sever reviewed (approved):
Remove, deleted by upstream  zebra`ip protocols connected|kernel|table` options
Add migration scripts
